mockgcp/mockcompute/networksv1.go

```diff
@@ -127,3 +129,34 @@ func (s *NetworksV1) Delete(ctx context.Context, req *pb.DeleteNetworkRequest) (
 
 	return s.newLRO(ctx, name.Project.ID)
 }
+
+type networkName struct {
+	Project *projects.ProjectData
+	Name    string
+}
+
+func (n *networkName) String() string {
+	return "projects/" + n.Project.ID + "/global" + "/networks/" + n.Name
+}
+
+// parseNetworkName parses a string into a networkName.
+// The expected form is `projects/*/global/networks/*`.
+func (s *MockService) parseNetworkName(name string) (*networkName, error) {
```

Comment:
Wondering why we want to intermediary name field for an internal function. Not seeing any signs of having it in single string form. All calling sites seem to already have the 2 fields already broken out and then we assemble the string so we can parse out those 2 fields.

Reply:
Similar to the other question, we do resolve the project and normalize it.

I am wondering if we should generate the parse and String functions, e.g. using tags on the struct, but that feels like (1) something AI could do and (2) "pre AI" thinking, i.e. it's better to have concrete code so the AI can operate on it (even if there's more of it), than to have complex meta programming that the AI has no idea about.